mlflow/store/_unity_catalog/registry/utils.py

```diff
@@ -106,3 +108,20 @@ def _get_artifact_repo_from_storage_info(
         raise MlflowException(
             f"Got unexpected token type {credential_type} for Unity Catalog managed file access"
         )
+
+
+def get_full_name_from_sc(name, spark) -> str:
+    """
+    Constructs the full name of a registered model using the active catalog and schema in a spark
+    session / context.
+    :param name: the model name provided by the user
+    :param spark: the active spark session
+    """
+    num_levels = len(name.split("."))
+    if num_levels >= 3 or spark == None:
+        return name
+    catalog = spark.sql(_ACTIVE_CATALOG_QUERY).collect()[0]["catalog"]
```

Comment:
What happens if we run the query on a non-UC enabled cluster, does it throw a weird error? Is that a problem?

Reply:
This might also throw if it gets run in a Spark job, we might want to catch and rethrow this with a nicer error message

Reply:
When I run it on a non-UC-enabled cluster, the queries work, except it will return `spark_catalog` as the catalog. For detecting this, I think we should have some high-level validation at the REST store instantiation time.

Reply:
Let me try to run this in a Spark Job.

Reply:
Added the Spark job testing results to the PR manual testing description.